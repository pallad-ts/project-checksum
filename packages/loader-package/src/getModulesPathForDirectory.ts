import * as path from "path";

const ROOT_PATHS = new Set(['.', '/']);

export function getModulesPathsForDirectory(directory: string) {
	let currentDirectory = directory;
	const paths = [] as string[]
	do {
		paths.push(path.join(currentDirectory, 'node_modules'));
		currentDirectory = path.dirname(currentDirectory);
	} while (!ROOT_PATHS.has(currentDirectory))
	paths.push('/node_modules');
	return paths;
}
