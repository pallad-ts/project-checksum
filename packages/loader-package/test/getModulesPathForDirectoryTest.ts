import {getModulesPathsForDirectory} from "@src/getModulesPathForDirectory";

describe('getModulesPathForDirectory', () => {
	it('walks up to root directory', () => {
		expect(getModulesPathsForDirectory('/root/projects/test/alpha/beta'))
			.toMatchSnapshot();
	});
});
