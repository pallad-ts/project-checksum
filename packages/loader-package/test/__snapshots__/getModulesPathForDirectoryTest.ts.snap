// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getModulesPathForDirectory walks up to root directory 1`] = `
[
  "/root/projects/test/alpha/beta/node_modules",
  "/root/projects/test/alpha/node_modules",
  "/root/projects/test/node_modules",
  "/root/projects/node_modules",
  "/root/node_modules",
  "/node_modules",
]
`;
